internal/metrics/recorder.go

```diff
@@ -83,7 +95,56 @@ func NewRecorder(register bool) *Recorder {
 		addonsCount:           addonsCount,
 		addonOperatorPaused:   addonOperatorPaused,
 		ocmAPIRequestDuration: ocmAPIReqDuration,
+		addonHealthInfo:       addonHealthInfo,
+	}
+}
+
+func (r *Recorder) RecordAddonHealthInfo(addonStatus addonsv1alpha1.AddonStatus,
+	addonName, csvName, clusterID string) {
+
+	var (
+		// Regex for separating version from addon name
+		versionRegExp, _ = regexp.Compile(".v[0-9]+.[0-9]+.[0-9]+")
+		// Start and end index of the version substring
+		versionIndex []int
+		version      = "0.0.0" // default version when CSV is unavailable / installing
+
+		// `healthStatus` defaults to unknown unless status conditions say otherwise
+		healthStatus = 2
+		healthReason = "Unknown"
+		healthCond   = meta.FindStatusCondition(addonStatus.Conditions,
+			addonsv1alpha1.Available)
+	)
+
+	if csvName != "" {
+		// The following lines separate the version and addonName from the CSV Name
+		// For example, in `reference-addon.v.1.1.1`, `addonName` is "reference-addon"
+		// and `version` is "1.1.1"
+		versionIndex = versionRegExp.FindStringIndex(csvName) // index of the version substring
+		version = csvName[versionIndex[0]+2:]                 // version
+	}
```

Comment:
I see why you have to do this, to report _some_ version before the CSV is present on the cluster, but I still don't like it, as it's making initial metrics very hard to aggregate.
There is also an issue, what version is actually meaning. I am not sure `csv version == addon version` stands always true, because, at least for a while, you can make changes to addons, which does not require to release a new bundle.

But what do you think about adding `version` to the Addon Spec and having OCM inject the "version" string that they are tracking? This way we will have the version always available for accurate reporting.